README.md

```diff
@@ -51,22 +51,30 @@ There are different ways of creating providers inside Pulumi but for this projec
 
 ## How to Use
```

Comment:
So in my PR, I actually trimmed the README file quite a bit and restructured the content: I put "How to Use" at the top then added a "Technology Stack" section where I left a brief description of Twilio, Pulumi, and your brief note on Dynamic Provider. What do you think about that, @kaiquelupo ? Maybe we can merge this PR first as I want to incorporate @gverni 's changes.